### PR TITLE
Improve command extensions

### PR DESCRIPTION
**Pull request by DRSchlaubi**
- Add AutoCompletingCommand
- Add test for AutoCompletingCommand
- Add EDSL for commands
- Update README to reflect new changes
- Add convince method to use ConfigurationSection#getSerializable() more easily

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -86,10 +97,10 @@ publishing {
             version version.toString()
             pom.withXml {
                 def root = asNode()
-                root.appendNode("name", "spiglin")
+                root.appendNode("name:spiglin")
                 root.appendNode("description",
                         "A collection of Kotlin extensions and utilities for the Spigot/Bukkit framework.")
-                root.appendNode("url", "https://github.com/johnnyjayjay/spiglin/")
+                root.appendNode("url:https://github.com/johnnyjayjay/spiglin/")
```
Comment (JohnnyJayJay) [inline build.gradle]: Is there a particular reason for these changes?

File: src/main/kotlin/com/github/johnnyjayjay/spiglin/command/CommandDSL.kt [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import org.bukkit.plugin.java.JavaPlugin
+
+/**
+ * Registers a [CommandExecutor] with [name] defined by [convention] to this plugin.
+ *
+ * @throws IllegalArgumentException if [JavaPlugin.getCommand] returned null
+ */
+fun JavaPlugin.command(name: String, convention: CommandConvention.() -> Unit) {
+    val command = getCommand(name)
+    requireNotNull(command) { "Command is null, did you forget to register it in your plugin.yml" }
```
Comment (JohnnyJayJay) [inline src/main/kotlin/com/github/johnnyjayjay/spiglin/command/CommandDSL.kt]: Add ? in the message here

File: README.md [hunk 3]
```diff
@@ -115,6 +115,60 @@ The children themselves can also be `DelegatingCommand`s of course.
...
 the previously first argument is dropped and the label is changed to the child label.
 
+There also is an Extension of `DelegatingCommand` called `AutoCompletingCommand` which provides tab-complete suggestions for its children, the usage is the same as for `DelegatingCommand`
+
+### Command DSL
+In addition to `DelegatingCommand` and `AutoCompletingCommand` there also is an EDSL providing an easier way to create more complex command structures.
+
+The `command` method returns an instance of an `CommandExecutor` that can be registered in the same way as an `DelegatingCommand` or `AutoCompletingCommand`
+
+Each individual command is defined as a [lambda with a receiver](https://kotlinlang.org/docs/reference/lambdas.html#function-literals-with-receiver) taking an instance of the `CommandContext class` which has the properties as the parameters of `CommandExecutor#onCommand()`
```
Comment (JohnnyJayJay) [inline README.md]: I'm not too sure about this one. I think it should be a regular parameter, not the receiver. You don't execute the command *on* the context, but *with* the context.

**Review by JohnnyJayJay — COMMENTED** [review]
Good work all in all. I think this can be merged for version 2.1 upon further testing by my side.

File: build.gradle (hunk 1 above)
Comment (DRSchlaubi) [reply build.gradle]: I don't even remember even touching those lines not sure why they're changed

File: src/main/kotlin/com/github/johnnyjayjay/spiglin/command/CommandDSL.kt (hunk 2 above)
Comment (DRSchlaubi) [reply src/main/kotlin/com/github/johnnyjayjay/spiglin/command/CommandDSL.kt]: Sure

File: build.gradle (hunk 1 above)
Comment (DRSchlaubi) [reply build.gradle]: Oh I think I copied a dependencyspec that looked like this implementation "group", "artifact", "version" and replaced ", " with :

**Review by JohnnyJayJay — APPROVED** [review]

**Comment by JohnnyJayJay** [comment]
Will be included with 2.1.

**Pull request merged**